src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,72 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final ColorSupplier color = new ColorSupplier();
+    private static final Random random = new Random();
+    private static final FigureList[] figures = FigureList.values();
+    private static final double RANDOM_MAX_VALUE = 5.1;
+    private static final double DEFAULT_CIRCLE_RADIUS = 10;
+
+    public Figure getRandomFigure() {
+        Figure figure = null;
+
+        switch (randomiseFigureType()) {
+            case CIRCLE:
+                figure = getCircle();
+                break;
+            case ISOSCELES_TRAPEZOID:
+                figure = getIsoscelesTrapezoidFigure();
+                break;
+            case RIGHT_TRIANGLE:
+                figure = getRightTriangle();
+                break;
+            case RECTANGLE:
+                figure = getRectangle();
+                break;
+            case SQUARE:
+                figure = getSquare();
+                break;
+            default:
+                getDefaultFigure();
+        }
+        return figure;
+    }
+
+    private Figure getSquare() {
+        return new Square(color.getRandomColor(),
+                random.nextDouble(RANDOM_MAX_VALUE));
+    }
+
+    private Figure getRectangle() {
+        return new Rectangle(color.getRandomColor(),
+                random.nextDouble(RANDOM_MAX_VALUE),
+                random.nextDouble(RANDOM_MAX_VALUE));
+    }
+
+    private Figure getRightTriangle() {
+        return new RightTriangle(color.getRandomColor(),
+                random.nextDouble(RANDOM_MAX_VALUE),
+                random.nextDouble(RANDOM_MAX_VALUE));
+    }
+
+    private Figure getIsoscelesTrapezoidFigure() {
+        return new IsoscelesTrapezoid(color.getRandomColor(),
+                random.nextDouble(RANDOM_MAX_VALUE),
+                random.nextDouble(RANDOM_MAX_VALUE),
+                random.nextDouble(RANDOM_MAX_VALUE));
+    }
+
+    public Figure getDefaultFigure() {
```

Comment:
```suggestion
    private Figure getDefaultFigure() {
```

Reply:
I invoke this method from Main class. Should i create public metod getFigure() in FigureSupplier that invoke private getDefaultFigure()?

Reply:
No, my mistake. 
Replace it to public methods. 
Public first - > private after